apps/meteor/ee/app/livechat-enterprise/server/lib/Helper.js

```diff
@@ -162,11 +162,15 @@ export const setPredictedVisitorAbandonmentTime = (room) => {
 	LivechatRooms.setPredictedVisitorAbandonment(room._id, willBeAbandonedAt);
 };
 
-export const updatePredictedVisitorAbandonment = () => {
+export const updatePredictedVisitorAbandonment = async () => {
 	if (!settings.get('Livechat_abandoned_rooms_action') || settings.get('Livechat_abandoned_rooms_action') === 'none') {
 		LivechatRooms.unsetPredictedVisitorAbandonment();
 	} else {
-		LivechatRoomsRaw.findPaginated({ t: 'l', open: true }).cursor.forEach((room) => setPredictedVisitorAbandonmentTime(room));
+		// Eng day: use a promise queue to update the predicted visitor abandonment time instead of all at once
+		const promisesArray = [];
+		LivechatRoomsRaw.find({ t: 'l', open: true }).forEach((room) => promisesArray.push(setPredictedVisitorAbandonmentTime(room)));
```

Comment:
I think you'll need to `await` this `forEach`, otherwise `promisesArray` will still be empty..

since you'll need to change this, may I ask you to also create a model method to perform this query? 😇